sql/hive/src/main/scala/org/apache/spark/sql/hive/orc/OrcRelation.scala

```diff
@@ -120,16 +110,33 @@ private[orc] class OrcOutputWriter(
     ).asInstanceOf[RecordWriter[NullWritable, Writable]]
   }
 
-  override def writeInternal(row: InternalRow): Unit = {
+  private def wrapOrcStruct(
+      o: OrcStruct,
+      oi: SettableStructObjectInspector,
+      a: InternalRow): OrcStruct = {
+    val fieldRefs = oi.getAllStructFieldRefs
+    val row = a.asInstanceOf[InternalRow]
     var i = 0
-    while (i < row.numFields) {
-      reusableOutputBuffer(i) = wrappers(i)(row.get(i, dataSchema(i).dataType))
+    while (i < fieldRefs.length) {
```

Comment:
`StandardStructObjectInspector` will implicitly lowercase column names. Otherwise, it uses `OrcStruct`. I think there are no other significant differences.